agent/agent_test.go

```diff
@@ -103,6 +104,58 @@ func TestReceiveLoop(t *testing.T) {
 	<-done
 }
 
+// TestPeriodicKeepalive checks that a running Agent sends its periodic
+// keepalive messages at the expected frequency, allowing for +/- 1s drift.
+// TODO: use Eric's crock package to test many more ticks in a short amount of time.
+func TestPeriodicKeepalive(t *testing.T) {
+	done := make(chan struct{})
+
+	cfg := FixtureConfig()
+	server := transport.NewServer()
+
+	testKeepalive := func(w http.ResponseWriter, r *http.Request) {
+		conn, err := server.Serve(w, r)
+		assert.NoError(t, err)
```

Comment:
This should be require.NoError, we can't continue the test if this fails.

Reply:
Of course!
Will change it.